### PR TITLE
Enable audio playback by implementing libcubeb backend for Haiku

### PR DESCRIPTION
**Pull request by kenz-gelsoft**

### CODE REVIEW

File: media/libcubeb/src/cubeb_haiku.cpp [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+      break;
+    case media_raw_audio_format::B_AUDIO_SHORT:
+      frames /= sizeof(int16_t);
+      break;
+    case media_raw_audio_format::B_AUDIO_CHAR:
+      frames /= sizeof(int8_t);
+      break;
+  }
+
+  long got = stm->data_callback(stm, stm->user_ptr, nullptr, buffer, frames);
```
Comment (kenz-gelsoft) [inline media/libcubeb/src/cubeb_haiku.cpp:151]: we need to bail out before this line, when receiving side was finished.

Comment (kenz-gelsoft) [reply media/libcubeb/src/cubeb_haiku.cpp:151]: It was not the case here. I have no idea to fix this now.

**Comment by waddlesplash** [comment]
Maybe we could upgrade the Cubeb recipe/patchset and use the system cubeb? https://github.com/haikuports/haikuports/tree/master/media-libs/cubeb

**Comment by kenz-gelsoft** [comment]
There is no option to use system libcubeb in build system currently. And libcubeb is about 4 years behind. If there is small or no change in API/ABI, it probably work if we make a change to firefox build system to support using system library for libcubeb.

For example, FreeBSD or other BSD port has a patch to use more system library (not libcubeb).

**Comment by kenz-gelsoft** [comment]
~~Just applying changes to haiku128.5 branch didn't work.~~ Edit: No. it worked as before just crashing after audio playback in debug build.

I'll rebase this branch and update changes.

**Comment by kenz-gelsoft** [comment]
> Edit: No. it worked as before just crashing after audio playback in debug build.

I worked around this in 9b96165390236684f1171c816709f5d33b10b30b.

Will investigate in #84.

I'll merge this for now.

**Pull request merged**